### ISSUE TITLE
specifying options on the command line

### ISSUE THREAD

**Issue by natechols** [PacificBiosciences/pbsmrtpipe#210]
In the category of nice-to-have-but-not-essential, I'd like to be able to provide arguments like this:

--option pbsmrtpipe.options.max_nproc:24

or:

--task-option pbalign.task_options.concordant:False

For quick testing (especially in parallel), this is much easier than editing XML files.

(I am, of course, happy to write the implementation myself if this idea is worthwhile.)

**Comment by mpkocher** [comment]
In general, I like the idea, however, pragmatically, I am not sure this would be the useful. The identifiers are so long that misspellings would be very common. 

The model that I use is to run `pbsmrtpipe show-template-details pbsmrtpipe.pipelines.sa3_sat -o sat-defaults.xml` for a specific pipeline, then update the value(s) in the XML. 

If you're using the align params in different pipelines, then maybe creating an handful of commonly used custom presets for align, (e.g.,  `align-unrolled-preset.xml`) would be reasonable work around.

The 'engine level options' I think could be useful, specifically if they're not prefixed with `pbsmrtpipe.options.`, such as  `--max-nproc=13`. However, you can do this same thing `pbsmrtpipe show-workflow-options -o default-engine-opt.xml`. 

Then run pbsmrtpipe with multiple `--preset-xml` values.
